src/tool.cpp

```diff
@@ -110,40 +107,42 @@ DigParams getDigParams(const ItemGroupList &groups,
 	std::string result_main_group = "";
 
 	int level = itemgroup_get(groups, "level");
-	//infostream<<"level="<<level<<std::endl;
 	for (ToolGCMap::const_iterator i = tp->groupcaps.begin();
 		 	i != tp->groupcaps.end(); ++i) {
-		const std::string &name = i->first;
-		//infostream<<"group="<<name<<std::endl;
 		const ToolGroupCap &cap = i->second;
-		int rating = itemgroup_get(groups, name);
+
+		int leveldiff = cap.maxlevel - level;
+		if (leveldiff < 0)
+			continue;
+
+		const std::string &groupname = i->first;
 		float time = 0;
-		bool time_exists = cap.getTime(rating, &time);
-		if(!result_diggable || time < result_time){
-			if(cap.maxlevel >= level && time_exists){
-				result_diggable = true;
-				int leveldiff = cap.maxlevel - level;
-				result_time = time / MYMAX(1, leveldiff);
-				if(cap.uses != 0)
-					result_wear = 1.0 / cap.uses / pow(3.0, (double)leveldiff);
-				else
-					result_wear = 0;
-				result_main_group = name;
-			}
+		bool time_exists = cap.getTime(itemgroup_get(groups, groupname), &time);
```

Comment:
It's not entirely clear what `getTime` returns, thus it's better to keep it this way. Makes it easier to understand and the compiler would optimize this anyway.